packages/core/src/client.ts

```diff
@@ -3875,22 +3885,25 @@ export class MedplumClient extends TypedEventTarget<MedplumClientEventMap> {
   /**
    * Makes a POST request to the tokens endpoint.
    * See: https://openid.net/specs/openid-connect-core-1_0.html#TokenEndpoint
-   * @param formBody - Token parameters in URL encoded format.
+   * @param params - Token parameters.
    * @returns The user profile resource.
    */
-  private async fetchTokens(formBody: URLSearchParams): Promise<ProfileResource> {
+  private async fetchTokens(params: Record<string, string>): Promise<ProfileResource> {
+    const formBody = new URLSearchParams(params);
+    const headers: HeadersInit = { 'Content-Type': ContentType.FORM_URL_ENCODED, ...this.defaultHeaders };
+    if (this.basicAuth) {
+      headers['Authorization'] = `Basic ${this.basicAuth}`;
+    } else if (this.credentialsInHeader && params.client_id && params.client_secret) {
+      headers['Authorization'] = `Basic ${encodeBase64(params.client_id + ':' + params.client_secret)}`;
+      formBody.delete('client_id');
+      formBody.delete('client_secret');
+    }
```

Comment:
What if basicAuth is defined, but also credentialsInHeader is true?

Based on the subsequent logic of the else, you might want to delete the client_id and client_secret from the body in that situation.

Perhaps something along the lines of

```suggestion
if (this.basicAuth) {
  headers['Authorization'] = `Basic ${this.basicAuth}`;
} 

if (this.credentialsInHeader) {
  formBody.delete('client_id');
  formBody.delete('client_secret');

  if ((!this.basicAuth) && (params.client_id && params.client_secret)) {
    headers['Authorization'] = `Basic ${encodeBase64(params.client_id + ':' + params.client_secret)}`;
  }
}
```